tests/sdk/utils.py

```diff
@@ -32,12 +32,13 @@ def mock_requests_response(response=None, status=200, headers=None):
     return m
 
 
-def mock_requests(mocker, method, response=None, status=200, headers=None):
-
-    def _req(*args, **kwargs):
-        return mock_requests_response(response, status, headers)
-
+def mock_requests_responses(mocker, method, responses):
     return mocker.patch(
         f'substra.sdk.rest_client.requests.{method}',
-        side_effect=_req,
+        side_effect=responses,
```

Comment:
It looks like there is only one use of this new function, but I see for example in `test_download.py` L72 that we used to do the same thing manually. I think we should either replace all mocks of requests using this function or none of them.